### PR TITLE
Fixed test failures when AE enabled

### PR DESCRIPTION
**Pull request by lilgreenbird**

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java [hunk 1]
```diff
@@ -224,7 +224,18 @@ public void testInvalidStoredProcedureName() throws SQLException {
                 pstmt.execute();
             } catch (SQLException e) {
                 if (!e.getMessage().contains(TestResource.getResource("R_StoredProcedureNotFound"))) {
-                    fail(e.getMessage());
+                    // diff error msg if AE enabled
+                    int start = connectionString.toUpperCase().indexOf(Constants.COLUMN_ENCRYPTION_SETTING);
```
Comment (rene-ye) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:228]: It's not a real concern as it's a scenario that'd happen if someone really wanted to break this logic, but our driver takes the last instance of the value. So if you declare columnEncryptionSetting=enabled;columnEncryptionSetting=disabled, this logic would detect it's on even though it's off.

Comment (ulvii) [reply src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:228]: Looks like you are trying to check if AE is enabled. You can simply use `ds.getColumnEncryptionSetting()`.

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java [hunk 2]
```diff
@@ -647,6 +648,17 @@ private void testStatementPoolingInternal(String mode) throws Exception {
                 assertNotSame(0, handle);
             }
 
+            // AE has 1 more prepared statement handle as it calls sp_describe_parameter_encryption
+            int start = connectionString.toUpperCase().indexOf(Constants.COLUMN_ENCRYPTION_SETTING);
+            if (-1 != start) {
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java:653]: Same here, use  `ds.getColumnEncryptionSetting()`

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java [hunk 3]
```diff
@@ -224,7 +224,18 @@ public void testInvalidStoredProcedureName() throws SQLException {
                 pstmt.execute();
             } catch (SQLException e) {
                 if (!e.getMessage().contains(TestResource.getResource("R_StoredProcedureNotFound"))) {
-                    fail(e.getMessage());
+                    // diff error msg if AE enabled
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:227]: Different error message if AE is enabled.

File: src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java [hunk 4]
```diff
@@ -570,7 +571,7 @@ public void testIllegalNumberOfArgNoColumnList() throws Exception {
 
             pstmt.executeBatch();
             fail(TestResource.getResource("R_expectedExceptionNotThrown"));
-        } catch (BatchUpdateException e) {
+        } catch (BatchUpdateException | SQLServerException e) {
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java:574]: Can you explain why you have to do this?

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java [hunk 5]
```diff
@@ -224,7 +224,18 @@ public void testInvalidStoredProcedureName() throws SQLException {
...
+                    int start = connectionString.toUpperCase().indexOf(Constants.COLUMN_ENCRYPTION_SETTING);
+                    if (-1 != start) {
+                        int end = connectionString.indexOf(";", start);
+                        String value = connectionString.substring(
+                                start + Constants.COLUMN_ENCRYPTION_SETTING.length() + 1,
+                                (-1 != end) ? end : connectionString.length());
+                        if (value.equalsIgnoreCase(Constants.ENABLED)
+                                && !e.getMessage().contains(TestResource.getResource("R_incorrectSyntaxP0"))) {
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:235]: Can you explain why this error message is thrown?

Comment (lilgreenbird) [reply src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:235]: that's the error returned by sql server when columns are encrypted

File: src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java (hunk 4 above)
Comment (lilgreenbird) [reply src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java:574]: fails thru a different code path for AE

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java [hunk 6]
```diff
@@ -223,12 +223,14 @@ public void testInvalidStoredProcedureName() throws SQLException {
                 pstmt.setStructured(1, tvpName, rs);
                 pstmt.execute();
             } catch (SQLException e) {
-                if (!e.getMessage().contains(TestResource.getResource("R_StoredProcedureNotFound"))) {
+                if (!e.getMessage().contains(TestResource.getResource("R_StoredProcedureNotFound"))
+                        && !(ds.getColumnEncryptionSetting().equalsIgnoreCase(Constants.ENABLED)
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java:227]: Will this ever pass ? Don't you need to `||` (or)?

**Pull request merged**